Source/WTF/wtf/text/StringImpl.h

```diff
@@ -1382,35 +1395,44 @@ inline Expected<std::invoke_result_t<Func, Span<const char>>, UTF8ConversionErro
     if (length > MaxLength / 3)
         return makeUnexpected(UTF8ConversionError::OutOfMemory);
 
-    if (is8Bit()) {
-        const LChar* characters = this->characters8() + offset;
 #if CPU(ARM64)
-        if (const LChar* nonASCII = find8NonASCII(characters, length)) {
-            size_t prefixLength = nonASCII - characters;
-            size_t remainingLength = length - prefixLength;
-            Vector<char, 1024> bufferVector(prefixLength + remainingLength * 3);
-            char* buffer = bufferVector.data();
-
-            memcpy(buffer, characters, prefixLength);
-            buffer += prefixLength;
-
-            auto success = Unicode::convertLatin1ToUTF8(&nonASCII, characters + length, &buffer, buffer + (bufferVector.size() - prefixLength));
-            ASSERT_UNUSED(success, success); // (length * 3) should be sufficient for any conversion
-            return function(Span<const char> { bufferVector.data(), buffer });
-        } else
-            return function(Span<const char> { bitwise_cast<const char*>(characters), bitwise_cast<const char*>(characters + length) });
-#else
-        Vector<char, 1024> bufferVector(length * 3);
+    if (const LChar* nonASCII = find8NonASCII(characters, length)) {
+        size_t prefixLength = nonASCII - characters;
+        size_t remainingLength = length - prefixLength;
+
+        Vector<char, 1024> bufferVector(prefixLength + remainingLength * 3);
         char* buffer = bufferVector.data();
-        auto success = Unicode::convertLatin1ToUTF8(&characters, characters + length, &buffer, buffer + bufferVector.size());
+
+        memcpy(buffer, characters, prefixLength);
+        buffer += prefixLength;
+
+        auto success = Unicode::convertLatin1ToUTF8(&nonASCII, characters + length, &buffer, buffer + (bufferVector.size() - prefixLength));
         ASSERT_UNUSED(success, success); // (length * 3) should be sufficient for any conversion
         return function(Span<const char> { bufferVector.data(), buffer });
-#endif
     }
+    return function(Span<const char> { bitwise_cast<const char*>(characters), bitwise_cast<const char*>(characters + length) });
+#else
+    Vector<char, 1024> bufferVector(length * 3);
+    char* buffer = bufferVector.data();
+    const LChar* source = characters;
+    bool success = Unicode::convertLatin1ToUTF8(&source, source + length, &buffer, buffer + bufferVector.size());
+    ASSERT_UNUSED(success, success); // (length * 3) should be sufficient for any conversion
+    return function(Span<const char> { bufferVector.data(), buffer });
+#endif
+}
 
+template<typename Func>
+inline Expected<std::invoke_result_t<Func, Span<const char>>, UTF8ConversionError> StringImpl::tryGetUtf8ForCharacters(const Func& function, const UChar* characters, unsigned length, ConversionMode mode)
+{
+    if (!length) {
+        const char* const emptyString = "";
+        return function(Span<const char> { emptyString, emptyString });
```

Comment:
Can we write just `Span` here or does it not deduce the type?

Reply:
Changed.